lib/util/reddit.ts

```diff
@@ -0,0 +1,68 @@
+import * as centra from "centra";
+import { randomInt } from "crypto";
+import { RedditPost } from "../interfaces/reddit";
+
+/* API */
+
+/**
+ * Returns a random post/meme from a subreddit.
+ * @param subreddit Optional The subreddit string, defaults to a random english meme subreddit.
+ * @returns {RedditPost}
+ */
+ export async function getRandomPost(subreddit?: string): Promise<RedditPost | undefined> {
+	const url = `https://www.reddit.com/r/${subreddit ? subreddit : randomSubreddit()}/hot/.json?count=100`
+
+	return await buildPost(url)
+}
+
+
+/* Utils */
+
+// more languages coming soontm
+const randomSubreddits = {
+  en: ['memes', 'dankmemes', 'meirl'],
+}
+
+/**
+ * Gets a random subreddit from the list of available meme subreddits.
+ * @returns {string} The random subreddit string.
+ */
+function randomSubreddit(): string {
+	return randomSubreddits.en[randomInt(randomSubreddits.en.length)]
+}
+
+
+export function checkURL(url: string): boolean {
+  return url.match(/\.(jpeg|jpg|gif|png)$/) != null
```

Comment:
```suggestion
  return /\.(jpeg|jpg|gif|png)$/.test(url)
```

Reply:
this could also be done entirely without regex;
```js
[].reduce((a, c) => a || url.endsWith(c), false)
```

Reply:
how do i thumbs down